packages/plugin-auto-nav-sidebar/src/utils.ts

```diff
@@ -28,16 +28,21 @@ export async function detectFilePath(rawPath: string) {
 export async function extractH1Title(
   filePath: string,
   rootDir: string,
+  /** prefer using the title declared in frontmatter rather than file content */
+  preferFrontMatter = true,
```

Comment:
Why we perfer using `frontmatter.title`? For performance, we used the  h1 title in origin content and fallback to the frontmatter.title when  h1 title does not exist.

Reply:
If using h1 title in the origin content, then in #534 's case, the sidebar will look like
<img width="1542" alt="image" src="https://github.com/web-infra-dev/rspress/assets/73387709/bf0229bb-0e99-4bbf-8eed-1e5e47182583">

That may not meet the developer's need

Reply:
All right.

Reply:
Maybe we can create a front matter cache manager in `@rspress/shared` to parse and reuse all the frontmatter info  of md(x) files to avoid duplicated parse process.

Reply:
> If using h1 title in the origin content, then in #534 's case, the sidebar will look like <img alt="image" width="1542" src="https://private-user-images.githubusercontent.com/73387709/298337594-bf0229bb-0e99-4bbf-8eed-1e5e47182583.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4MDg1OTYsIm5iZiI6MTcwNTgwODI5NiwicGF0aCI6Ii83MzM4NzcwOS8yOTgzMzc1OTQtYmYwMjI5YmItMGU5OS00YmJmLThlZWQtMWU1ZTQ3MTgyNTgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDAzMzgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkN2I4ZjA1MzVkNjA0ZTkxY2I4MTc4MzQ0MTFlMzNhMjg1YzZkMjlmNmUwMjM2ZDdlYzJjMjhlN2I1YmMyYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BJIXUJ4JpsHtpXjTcBgBtI1S3M7iKyawFIyS5bsGHIM">
> 
> That may not meet the developer's need

Or this case shows a wrong usage? If so, we should not prefer using `frontmatter.title`

Reply:
yeah, i think it shows a unexpected usage.